src/appleseed/foundation/utility/iesparser.cpp

```diff
@@ -0,0 +1,946 @@
+    bool is_negative(const ValueType& type)
+    {
+        return type < ValueType(0);
+    }
+
+    // Lexical cast without exception.
+    template <typename ValueType>
+    bool string_to_number(const std::string& str, ValueType* value)
+    {
+        try
+        {
+            *value = boost::lexical_cast<ValueType>(str);
+        }
+        catch (boost::bad_lexical_cast&)
+        {
+            return false;
+        }
+        return true;
+    }
+
+    // Convert string to positive number.
+    template <typename ValueType>
+    bool string_to_positive_number(const std::string& str, ValueType* value)
+    {
+        try
+        {
+            *value = boost::lexical_cast<ValueType>(str);
+        }
+        catch (boost::bad_lexical_cast&)
+        {
+            return false;
+        }
+        if (*value <= ValueType(0))
+        {
+            return false;
+        }
+        return true;
+    }
+
+    // Convert string to photometric type.
+    bool string_to_photometric_type(
+        const std::string& str, IESParser::PhotometricType* type)
+    {
+        if (str == "1")
+        {
+            *type = IESParser::PhotometricTypeC;
+        }
+        else if (str == "2")
+        {
+            *type = IESParser::PhotometricTypeB;
+        }
+        else if (str == "3")
+        {
+            *type = IESParser::PhotometricTypeA;
+        }
+        else
+        {
+            return false;
+        }
+        return true;
+    }
+
+    // Convert string to units type.
+    bool string_to_units_type(
+        const std::string& str, IESParser::LuminousOpeningShape::UnitsType* type)
+    {
+        if (str == "1")
+        {
+            *type = IESParser::LuminousOpeningShape::Feet;
+        }
+        else if (str == "2")
+        {
+            *type = IESParser::LuminousOpeningShape::Meters;
+        }
+        else
+        {
+            return false;
+        }
+        return true;
+    }
+
+    // Split string and convert it to the vector of values.
+    // Returns false, if one of the tokens cannot be converted 
+    // to the value of specified type.
+    template <typename ValueType>
+    bool string_to_vector(
+        const std::string& str, std::vector<ValueType>* output)
+    {
+        std::vector<std::string> tokens;
+        boost::split(tokens, str, isspace, boost::token_compress_on);
+
+        output->clear();
+        output->reserve(tokens.size());
+
+        for (each< std::vector<std::string> > token = tokens; token; ++token)
+        {
+            try
+            {
+                output->push_back(boost::lexical_cast<ValueType>(*token));
+            }
+            catch (boost::bad_lexical_cast&)
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    template <typename ValueType>
+    bool check_increasing_order(const std::vector<ValueType>& values)
+    {
+        for (int i = 1; i < values.size(); ++i)
```

Comment:
You need to use `size_t` for the type of `i` here, this code won't compile with VS (which will complain about sign/unsigned comparison). Somehow this warning is disabled with gcc/clang in our builds, at some point we should look into why.